tests/models/cogvlm/test_modeling_cogvlm.py

```diff
@@ -0,0 +1,256 @@
+        model = CogVLMForCausalLM(config).to(torch_device).eval()
+        with torch.no_grad():
+            result = model(
+                pixel_values=pixel_values,
+                input_ids=input_ids,
+                attention_mask=attention_mask,
+                token_type_ids=token_type_ids,
+            )
+
+        self.parent.assertEqual(result.logits.shape, (self.batch_size, self.seq_length, self.vocab_size))
+
+    def prepare_config_and_inputs_for_common(self):
+        config_and_inputs = self.prepare_config_and_inputs()
+        (
+            config,
+            input_ids,
+            attention_mask,
+            token_type_ids,
+            pixel_values,
+            _,
+        ) = config_and_inputs
+        inputs_dict = {
+            "pixel_values": pixel_values,
+            "input_ids": input_ids,
+            "attention_mask": attention_mask,
+            "token_type_ids": token_type_ids,
+        }
+        return config, inputs_dict
+
+
+@require_torch
+class CogVLMModelTest(ModelTesterMixin, PipelineTesterMixin, unittest.TestCase):
+    all_model_classes = (CogVLMForCausalLM, CogVLMModel) if is_torch_available() else ()
+    pipeline_model_mapping = (
+        {"feature-extraction": CogVLMModel, "image-to-text": CogVLMForCausalLM} if is_torch_available() else {}
+    )
+
+    fx_compatible = False
+    test_head_masking = False
+    test_pruning = False
+    test_resize_embeddings = False
+    test_torchscript = False
+
+    def setUp(self):
+        self.model_tester = CogVLMModelTester(self)
+
+    def _prepare_for_class(self, inputs_dict, model_class, return_labels=False):
+        inputs_dict = copy.deepcopy(inputs_dict)
+        if return_labels:
+            if model_class.__name__ == "CogVLMForCausalLM":
+                inputs_dict["labels"] = torch.zeros(
+                    (self.model_tester.batch_size, self.model_tester.seq_length), dtype=torch.long, device=torch_device
+                )
+
+        return inputs_dict
+
+    def test_for_causal_lm(self):
+        config_and_inputs = self.model_tester.prepare_config_and_inputs()
+        self.model_tester.create_and_check_for_causal_lm(*config_and_inputs)
+
+    @unittest.skip(reason="Does not work on the tiny model as we keep hitting edge cases.")
+    def test_cpu_offload(self):
+        pass
+
+    @slow
+    def test_model_from_pretrained(self):
+        for model_name in COGVLM_PRETRAINED_MODEL_ARCHIVE_LIST:
+            model = CogVLMForCausalLM.from_pretrained(model_name)
+            self.assertIsNotNone(model)
+
+
+# We will verify our results on an image of cute cats
+def prepare_img():
+    url = "https://huggingface.co/hf-internal-testing/blip-test-image/resolve/main/demo.jpg"
+    image = Image.open(requests.get(url, stream=True).raw)
+    return image
+
+
+@require_vision
+@require_torch
+@slow
+class CogVLMModelIntegrationTest(unittest.TestCase):
```

Comment:
does interleaved images / text work? We need to test this as well. Also what are the token type ids used for? Need a test for this if there is a usecase

Reply:
No the model does not support interleaved images/text.

Token type ids are used to indicate text tokens (id = 0). The model adds token type ids for the vision tokens (id = 1) internally.

Reply:
alright let' s just add one example of explicit token type ids would be very nice to have here :)